### DISCUSSION TITLE
[WEB-881] Make all run flags available as turbo.json config options

### DISCUSSION THREAD

**Discussion by AlexAegis** [vercel/turborepo#7476] [Ideas]
### Which project is this feature idea for?

Turborepo

### Describe the feature you'd like to request

In my monorepo I have a lot of npm scripts invoking turbo. In every one of them I have to define my run options like `--concurrency 6 --cache-dir .cache/turbo`. I'd like to configure these flags only once, in `turbo.json`.

But the CLI options would still take precedence if I chose to override these default options.

### Describe the solution you'd like

In `turbo.json`, I'd like to define default run options like this:

```json
{
	"$schema": "https://turborepo.org/schema.json",
	"defaultOptions": {
		"concurrency": 6,
		"cacheDir": ".cache/turbo"
	},
	"pipeline": {
		...
	}
}
```

### Describe alternatives you've considered

Manually defining my default flags, but this is error-prone.

<sub>[WEB-881](https://linear.app/vercel/issue/WEB-881/make-all-run-flags-available-as-turbojson-config-options)</sub>

<sub>[TURBO-1142](https://linear.app/vercel/issue/TURBO-1142/[web-881]-make-all-run-flags-available-as-turbojson-config-options)</sub>

**Comment by RazeiXello** [comment]
Exactly what I was looking for today. Mainly to make [log-order](https://turbo.build/repo/docs/reference/command-line-reference/run#--log-order) grouped by default. +1

**Comment by JonasDoesThings** [comment]
Would also be useful for adding --continue to linting by default.

**Comment by RazeiXello** [comment]
To add to this, it would be great to be able to[ opt out of telemetry](https://turbo.build/repo/docs/telemetry#how-do-i-opt-out) via this defaultOptions property

```json
{
	"$schema": "https://turborepo.org/schema.json",
	"defaultOptions": {
		"disableTelemetry": true
	},
	"pipeline": {
		...
	}
}
```
